src/coreComponents/constitutive/solid/PorousSolid.hpp

```diff
@@ -285,33 +302,68 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
                                      q,
                                      timeIncrement,
                                      strainIncrement,
-                                     totalStress, // first effective stress increment accumulated
+                                     totalStress, // Rock stress updated with small strain
                                      stiffness );
 
-    // Add the contributions of pressure and temperature to the total stress
-    real64 const biotCoefficient = m_porosityUpdate.getBiotCoefficient( k );
-    real64 const thermalExpansionCoefficient = m_solidUpdate.getThermalExpansionCoefficient( k );
+    // Add the contributions of temperature increment to the rock stress (effective stress).
+    string const drainedTECTableName = m_solidUpdate.getDrainedTECTableName();
+    real64 dThermalExpansionCoefficient_dTemperature;
+    real64 thermalExpansionCoefficient;
+
+    if( drainedTECTableName.empty() )
+    {
+      real64 const defaultThermalExpansionCoefficient = m_solidUpdate.getThermalExpansionCoefficient( k );
+      real64 const referenceTemperature = m_solidUpdate.getReferenceTemperature();
+
+      dThermalExpansionCoefficient_dTemperature = m_solidUpdate.getDThermalExpansionCoefficient_dTemperature();
+      thermalExpansionCoefficient  = defaultThermalExpansionCoefficient + dThermalExpansionCoefficient_dTemperature * (temperature - referenceTemperature);
+    }
+    else
+    {
+      FunctionManager & functionManager = FunctionManager::getInstance();
+      TableFunction & drainedLinearTECTable = functionManager.getGroup< TableFunction >( drainedTECTableName );
+      drainedLinearTECTable.setInterpolationMethod( TableFunction::InterpolationType::Linear );
+      TableFunction::KernelWrapper TECWrapper = drainedLinearTECTable.createKernelWrapper();
```

Comment:
This cannot be done inside this DEVICE function. You'll have to create the Table wrapper object outside of the kernel launch call and then use it in here. 

You'll have to basically create that `kernelWrapper` for the table at the same time the `PorousSolidUpdates` object is crated.

Reply:
Thank you Mateo, this is an excellent point.

Reply:
Done